homeassistant/components/deconz/__init__.py

```diff
@@ -60,21 +60,29 @@ async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> b
     return True
 
 
-async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+async def async_unload_entry(
+    hass: HomeAssistant, config_entry: DeconzConfigEntry
+) -> bool:
     """Unload deCONZ config entry."""
-    hub: DeconzHub = hass.data[DOMAIN].pop(config_entry.entry_id)
+    hub = config_entry.runtime_data
     async_unload_events(hub)
 
-    if hass.data[DOMAIN] and hub.master:
+    other_loaded_entries: list[DeconzConfigEntry] = [
+        e
+        for e in hass.config_entries.async_loaded_entries(DOMAIN)
+        # exclude the config entry being unloaded
+        if e.entry_id != config_entry.entry_id
+    ]
+    if other_loaded_entries and hub.master:
         await async_update_master_hub(hass, config_entry)
```

Comment:
I think there is still a problem with the logic in `async_update_master_hub`

The previous logic used `hass.data` and the unloaded entry was already removed from `hass.data` at this point (despite still being in LOADED status)

In the new logic we rely on `runtime_data`, and it hasn't been cleared yet.

Reply:
I found a solution for this in bd002d655567a59f4f01b42d1416b1fb4a037493

Reply:
Im happy you didn't stay with "ass master" i couldnt think about much other than the south park movie when I initially read the property name 😂